### PR TITLE
Github actions for template

### LINKED ISSUE #169: Template is not using new matrix github action.

### PR DESCRIPTION
**Pull request by pdmthorsrud**
closes #169

### CODE REVIEW

**Comment by pdmthorsrud** [comment]
These github-workflows works so that when someone makes a new repo from template, they will have an integrated suite of github actions that are run on PR and push to master just like the vagrantbox's tests, and will try and run a playbook from `ansible/` in the template.

**Comment by pdmthorsrud** [comment]
Link to a working template using this code:
https://github.com/pdmthorsrud/testtemplate/actions

**Comment by fredrikhgrelland** [comment]
> These github-workflows works so that when someone makes a new repo from template, they will have an integrated suite of github actions that are run on PR and push to master just like the vagrantbox's tests, and will try and run a playbook from `ansible/` in the template.
> 
> Has anyone got an idea of how to avoid duplicating the `create_env.py` and linter files?

No, and we can live with that :+1:

File: template/.github/workflows/on_pr_push_master.yml [hunk 1]
```diff
@@ -1,14 +1,79 @@
-name: CI
-
+name: CI/CD
+env:
+  GITHUB_TOKEN: ${{ secrets.PAT_GITHUB_TOKEN }}
```
Comment (fredrikhgrelland) [inline template/.github/workflows/on_pr_push_master.yml:3]: Not necessary for template. Only needed in parent repo to push to template repo.
```suggestion
```

File: template/Makefile [hunk 2]
```diff
@@ -22,6 +22,14 @@ remove-tmp:
 
 clean: destroy-box remove-tmp
 
+test:
+ifeq (,$(wildcard ansible/playbook.yml))
+	echo ERROR: There is no playbook.yml available in ansible/ && exit 2
```
Comment (fredrikhgrelland) [inline template/Makefile:27]: nice :+1:

File: template/.github/workflows/on_pr_push_master.yml (hunk 1 above)
Comment (pdmthorsrud) [reply template/.github/workflows/on_pr_push_master.yml:3]: Whoops, thanks!

File: template/.github/workflows/on_pr_push_master.yml [hunk 3]
```diff
@@ -1,14 +1,77 @@
...
+        env:
+          test_nomad_acl: ${{ matrix.nomad_acl }}
+
+      - name: Make .env_override in order to matrix test
+        run: python .github/action/create-env.py
+        env:
+          ENVKEY_consul_acl: ${{ steps.consul_acl.outputs.consul_acl }}
+          ENVKEY_consul_acl_default_policy: ${{ steps.consul_acl_default_policy.outputs.default_policy }}
+          ENVKEY_nomad_acl: ${{ steps.nomad_acl.outputs.nomad_acl }}
+          FILE_NAME: .env_override
```
Comment (Neha-Sinha2305) [inline template/.github/workflows/on_pr_push_master.yml:74]: Shouldn't this be: **example/.env_override** ?
Have we run the github actions from inside the template and checked that things work as expected?

Comment (pdmthorsrud) [reply template/.github/workflows/on_pr_push_master.yml:74]: > Shouldn't this be: **example/.env_override** ?
> Have we run the github actions from inside the template and checked that things work as expected?

These actions are meant to test the code in template/ (which is nonexistent in this repo, but will be populated by a developer), and not in example/. I populated template/ and tested these actions in https://github.com/pdmthorsrud/testtemplate/actions

**Pull request merged**